qubes/tests/integ/vm_qrexec_gui.py

```diff
@@ -436,15 +436,32 @@ def wait_for_pulseaudio_startup(self, vm):
         # and some more...
         self.loop.run_until_complete(asyncio.sleep(1))
 
-    def prepare_audio_vm(self):
+    def prepare_audio_vm(self, backend):
         if 'whonix-gw' in self.template:
             self.skipTest('whonix-gw have no audio')
         self.loop.run_until_complete(self.testvm1.start())
+        pulseaudio_units = 'pulseaudio.socket pulseaudio.service'
+        pipewire_units = 'pipewire.socket wireplumber.service pipewire.service'
+        if backend == 'pulseaudio':
+            stop_units = pipewire_units
+            start_units = pulseaudio_units
+            manage_cmd = 'rm -f'
+        elif backend == 'pipewire':
+            start_units = pipewire_units
+            stop_units = pulseaudio_units
+            manage_cmd = 'touch'
+        else:
+            self.fail('bad audio backend')
         try:
-            self.loop.run_until_complete(
-                self.testvm1.run_for_stdio('which parecord'))
+            self.loop.run_until_complete("which parecord")
```

Comment:
`parecord` is executed in VM, not only in dom0, so the check should remain as it was. The check for dom0 is done via test decorator.